fury/ui.py

```diff
@@ -69,6 +69,53 @@ def clip_overflow(textblock, width, side='right'):
             return textblock.message
 
 
+def wrap_overflow(textblock, wrap_width):
+    """Wraps overflowing text of TextBlock2D with respect to width.
+
+    Parameters
+    ----------
+    textblock : TextBlock2D
+        The textblock object whose text needs to be wrapped.
+    wrap_width : int
+        Required width of the wrapped text.
+
+    Returns
+    -------
+    wrapped text : str
+        Wrapped version of the text.
+    """
+    original_str = textblock.message
+    start_ptr = 0
+    end_ptr = len(original_str)
+
+    if wrap_width <= 0:
+        raise ValueError(
+            f'Value of wrap_width cannot be less that or equal to 0'
+        )
+
+    if textblock.size[0] <= wrap_width:
+        return original_str
+
+    while start_ptr < end_ptr:
+        mid_ptr = (start_ptr + end_ptr)//2
+        textblock.message = original_str[:mid_ptr]
+        if textblock.size[0] < wrap_width:
+            start_ptr = mid_ptr
+        elif textblock.size[0] > wrap_width:
+            end_ptr = mid_ptr
+
+        if mid_ptr == (start_ptr + end_ptr)//2 or\
+           textblock.size[0] == wrap_width:
```

Comment:
You need to rework this, There's a a lot of code duplication here with respect to `clip_overflow`. 
That's why asked you to implement a separate method to check for overflow and then call it in `clip_overflow` and `wrap_overflow` as necessary.

Reply:
Hey @Nibba2018, sure this makes sense. Will update the PR accordingly.